src/ILCompiler.Compiler/src/Compiler/Compilation.cs

```diff
@@ -75,7 +75,7 @@ public abstract class Compilation : ICompilation
 
         private ILProvider _methodILCache;
         
-        public MethodIL GetMethodIL(MethodDesc method)
+        public virtual MethodIL GetMethodIL(MethodDesc method)
```

Comment:
I would rather not make this virtual. If all we need is for this to return null, could you just ifdef out the place that calls it?

(This is a place where we'll want an extension point in the future, but the extension point should be about allowing different IL providers, not about overriding this method. Having different IL providers is a concept that would be useful for all our backends (e.g. one could have an IL provider that provides pre-optimized IL that all backends could use). Having this also overridable would just be confusing.)

Reply:
I actually wonder whether this problem shouldn't go away at least partially with your latest change - I believe that the bugs I was looking at and trying to fix with this were caused by the callsite intrinsics that you've fixed directly in CorInfoImpl. I'll rerun the testing with the inclusion of your change and without my last change and I'll see where we land.